File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    
    bool check = false;
    for(int i = 0; i<s.size(); i++){
        if(s[i] == ' '){
            answer += ' ';
            check = false;
        }
        else if(!check){
            if(s[i] >= 97 && s[i] <= 122){
                answer += (s[i] - 32);
            }
            else answer += s[i];
            check = true;
        }
        else {
            if(s[i] >= 65 && s[i] <= 90)
                answer += (s[i] + 32);
            else answer += s[i];
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
class Solution {
    public String solution(String s) {
        StringBuilder sb = new StringBuilder();
        
        boolean isFirst = true;
        for(char c : s.toCharArray()) {
            if(isFirst) {
                sb.append(Character.toUpperCase(c));
            }
            else {
                sb.append(Character.toLowerCase(c));
            }
            
            isFirst = (c == ' ');
        }

        return sb.toString();
    }
}